### PR TITLE
Implement ECS `delete_task_definition` and `status` filter for `list_task_definitions`

### PR DESCRIPTION
**Pull request by lasuillard**
While I'm working on my personal project, I found there are some features missing in ECS that I need. So I implemented:

- I found that current `list_task_definitions` does not support `status` filter. I implement it added tests for it.
-The `delete_task_definitions` feature is not implemented, so I implemented it.

I have some questions:

- Because I thought `test_ecs_boto3.py` file quite big, I added tests for `delete_task_definitions` in new file, `test_ecs_task_definitions.py`. If I should merge the file, please let me know.

Thank you for maintaining this library. Please feel free to suggest any changes!

### CODE REVIEW

File: moto/ecs/models.py [hunk 1]
```diff
@@ -2445,5 +2463,66 @@ def enable_long_arn_for_name(self, name: str) -> bool:
...
+
+        deleted_task_definitions = []
+        failures = []
+        for task_def in task_definitions:
+            task_definition_name = task_def.split("/")[-1]
+            if ":" in task_definition_name:
+                family, rev = task_definition_name.split(":")
+                revision = int(rev)
+            else:
+                failures.append(
```
Comment (bblommers) [inline moto/ecs/models.py:2487]: @lasuillard CoPilot is right about the duplicate tests. 

Did you mean to test this edge case, so with a `task_definition_name` without a `:`?

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/ecs/models.py (hunk 1 above)
Comment (lasuillard) [reply moto/ecs/models.py:2487]: @bblommers You are right. That's what I exactly meant to do. Previously I manually checked the response of `aws ecs delete-task-definitions` and found that it gives me different response (`reason`) for each cases, so I wrote two very similar tests, `test_delete_task_definitions_invalid_identifier` and `test_delete_task_definitions_nonexistent`.

But now I see the code and the `reason` is same.. I must have made mistake. Thanks for pointing it out. Let me check it again and fix the code.

Comment (lasuillard) [reply moto/ecs/models.py:2487]: I pushed new commit to update the test response.
